AK/Time.cpp

```diff
@@ -54,6 +57,81 @@ Time Time::from_timespec(const struct timespec& ts)
     return Time::from_half_sanitized(ts.tv_sec, extra_secs, nsecs);
 }
 
+#ifndef KERNEL
+ErrorOr<Time> Time::from_iso_8601_timestamp(StringView iso_8601_timestamp)
+{
+    // Parse YYYY-MM-DDThh:mm:SS[.frac][tz] or YYYY-MM-DDThh:mm:SS[,frac][tz]
+    // formats, where brackets signify optional parameters.
+    GenericLexer lexer(iso_8601_timestamp);
+    int year;
+    unsigned month;
+    unsigned day;
+    unsigned hour;
+    unsigned minute;
+    unsigned second;
+    unsigned millisecond;
+    StringView time_zone;
+
+    auto lex_number = [&](size_t n) -> ErrorOr<int> {
+        auto maybe_value = lexer.consume(n).to_int();
+        if (!maybe_value.has_value())
+            return Error::from_string_view_or_print_error_and_return_errno("Invalid datetime format: expected number"sv, EINVAL);
```

Comment:
The reason I added this is that I originally wanted the function to be Kernel compatible, so this was very intentional, but then I figured out that ctypes is not accessible in the Kernel. We could add a Kernel version of ctypes (since it's basically just a table of ASCII character types, after all), but I didn't want to stack the yaks too high with this PR so I left it. I will put in a FIXME for Kernel compatibility of this function.